### PR TITLE
[MUSCLE] add recipe

### PR DESCRIPTION
**Pull request by M-PERSIC**
[MUSCLE](https://github.com/rcedgar/muscle) v5.2 (~~MacOS~~, Linux builds)

EDIT: Will attempt MacOS build in another pull request.

### CODE REVIEW

File: M/MUSCLE/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+                  "6c601163998616bb88991931e443c645858e162c"),
+    DirectorySource("./bundled"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd ${WORKSPACE}/srcdir/muscle/src/
+atomic_patch ${WORKSPACE}/srcdir/patches/remove-ffast-math.patch
+make -j${nproc}
+install -D -m 755 ${WORKSPACE}/srcdir/muscle/src/Linux/./muscle ${bindir}/./muscle
```
Comment (giordano) [inline M/MUSCLE/build_tarballs.jl:17]: Make it verbose so we have a record of what is being moved where
```suggestion
install -Dvm 755 "Linux/muscle" "${bindir}/muscle"
```

File: M/MUSCLE/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    ExecutableProduct("muscle", :muscle),
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[
+    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae")),
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6", preferred_gcc_version=v"11")
```
Comment (giordano) [inline M/MUSCLE/build_tarballs.jl:37]: Please read https://github.com/JuliaPackaging/Yggdrasil/blob/1ff4efc53e258ac8efd28875b40772e5989fffab/CONTRIBUTING.md#compatibility-tips carefully and after that pick an older version of GCC.

File: M/MUSCLE/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+cd ${WORKSPACE}/srcdir/muscle/src/
+atomic_patch ${WORKSPACE}/srcdir/patches/remove-ffast-math.patch
+make -j${nproc}
+install -D -m 755 ${WORKSPACE}/srcdir/muscle/src/Linux/./muscle ${bindir}/./muscle
+install_license ${WORKSPACE}/srcdir/muscle/LICENSE
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = filter!(p -> Sys.islinux(p), supported_platforms())
```
Comment (giordano) [inline M/MUSCLE/build_tarballs.jl:23]:
```suggestion
platforms = filter!(Sys.islinux, supported_platforms())
```

File: M/MUSCLE/build_tarballs.jl (hunk 2 above)
Comment (M-PERSIC) [reply M/MUSCLE/build_tarballs.jl:37]: The OSX build for MUSCLE [will require gcc11 at a minimum](https://github.com/rcedgar/muscle/wiki/Building-MUSCLE#building-under-osx).

Comment (giordano) [reply M/MUSCLE/build_tarballs.jl:37]: We don't use gcc on macOS.

Comment (M-PERSIC) [reply M/MUSCLE/build_tarballs.jl:37]: Okay. I'll tweak the Linux-only build recipe for now.

File: M/MUSCLE/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    GitSource("https://github.com/rcedgar/muscle.git",
+                  "6c601163998616bb88991931e443c645858e162c"),
+    DirectorySource("./bundled"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd ${WORKSPACE}/srcdir/muscle/src/
+atomic_patch ${WORKSPACE}/srcdir/patches/remove-ffast-math.patch
+make -j${nproc}
```
Comment (giordano) [inline M/MUSCLE/build_tarballs.jl:16]:
```suggestion
make -j${nproc} CXXFLAGS="-O3 -fopenmp"
```
is probably easier than patching the file.

Comment (M-PERSIC) [reply M/MUSCLE/build_tarballs.jl:16]: Ah thank you!

**Pull request merged**